wicket/src/ui/widgets/rack.rs

```diff
@@ -0,0 +1,365 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
This is primarily a moved file and doesn't require review. There are two major changes however: 
 * Resizing is no longer an event, and so the resize method of `RackState` is no a function that operates during render.
 *  Knight rider mode was simplified dramatically since we don't have Resize events anymore. This is actually a vast improvement :)